src/storage/disk_manager.c

```diff
@@ -5917,20 +5880,23 @@ disk_dump_goodvol_all (THREAD_ENTRY * thread_p, INT16 volid, void *arg)
 STATIC_INLINE bool
 disk_is_valid_volid (VOLID volid)
 {
-  // a passive transaction server (which is, implicitly, a transaction server
-  // with remote storage) maintains a separate bookkeeping of permanent data volumes
-  // which is updated when executing corresponding recovery replication functions
-  // as part of the regular replication; this way, it is able to provide a valid
-  // answer via this function
-  const bool is_valid_perm_volid = is_tran_server_with_remote_storage ()?
-    (volid < disk_Page_server_perm_volume_count) : (volid < disk_Cache->nvols_perm);
-  if (is_valid_perm_volid)
+  if (is_tran_server_with_remote_storage () && is_passive_transaction_server ())
     {
-      return is_valid_perm_volid;
+      /* Since PTS do not have a disk header page for newly added volume at the time of replication,
+       * disk_rv_redo_format (), which requires disk header page and does update the disk_Cache,
+       * is not called during replication. Thereefore, PTS knows only temporary volumes.
+       * Only the temporary volume ID is checked, and if it is not a temporary volume, only the range of volume id is checked */
+
+      const bool is_valid_temp_volid = (volid > (LOG_MAX_DBVOLID - disk_Cache->nvols_temp));
+      if (is_valid_temp_volid)
+	{
+	  return is_valid_temp_volid;
+	}
+
+      return volid >= LOG_DBFIRST_VOLID && volid <= LOG_MAX_DBVOLID;
```

Comment:
Whether `is_valid_temp_volid` is true or not, this is always true, if the volid is in the valid range. It seems to me that the `is_valid_temp_volid` does nothing.

Reply:
OK, I understand. we don't need to check `is_valid_temp_volid`, because the condition of checking the range of volume id will cover it.